packages/stats/storage.go

```diff
@@ -30,43 +30,55 @@ func NewStorage(l log.Logger, cfg config.ReplayorConfig) (Storage, error) {
 type Stats interface {
 	RecordBlockStats(bcs BlockCreationStats)
 	Write(ctx context.Context)
+	GetLastBlockWritten() uint64
 }
 
 type NoOpStats struct{}
 
 func (n *NoOpStats) RecordBlockStats(bcs BlockCreationStats) {}
 func (n *NoOpStats) Write(ctx context.Context)               {}
+func (n *NoOpStats) GetLastBlockWritten() uint64             { return uint64(0) }
 
-func NewStoredStats(s Storage, l log.Logger) Stats {
+func NewStoredStats(s Storage, l log.Logger, startBlock uint64) Stats {
 	return &StoredStats{
-		s:     s,
-		stats: []BlockCreationStats{},
-		log:   l,
+		s:                s,
+		stats:            []BlockCreationStats{},
+		log:              l,
+		lastBlockWritten: startBlock,
 	}
 }
 
 type StoredStats struct {
-	s     Storage
-	stats []BlockCreationStats
-	log   log.Logger
+	s                Storage
+	stats            []BlockCreationStats
+	log              log.Logger
+	lastBlockWritten uint64
+}
+
+func (s *StoredStats) GetLastBlockWritten() uint64 {
+	return s.lastBlockWritten
 }
```

Comment:
This change and the others in this file allow the `replayor` to periodically write results to storage. Currently writing every 100 blocks to storage. This is useful for a long-running test that gets interrupted unexpectedly. So instead of losing all data held in memory, you would only lose up to the last 100 blocks worth of data

Reply:
Maybe a better way to do it is catch any errors and ensure any data held in memory is written as a final cleanup step before shutting down